dcpy/utils/git.py

```diff
@@ -1,29 +1,63 @@
 import os
 import subprocess
+from typing import Optional
 
 
-def git_username():
+def username() -> Optional[str]:
     if os.environ.get("CI"):
         return os.environ.get("GITHUB_ACTOR_ID", "CI")
     else:
-        return (
-            subprocess.run(["git", "config", "user.name"], stdout=subprocess.PIPE)
-            .stdout.strip()
-            .decode()
-        )
+        try:
+            return (
+                subprocess.run(["git", "config", "user.name"], stdout=subprocess.PIPE)
+                .stdout.strip()
+                .decode()
+            )
+        except:
+            return None
 
 
-def git_branch():
+def branch() -> Optional[str]:
     if os.environ.get("CI"):
         ## REF_NAME - for push/workflow_dispatch
         ## HEAD_REF - for pull request triggers
         return os.environ.get("GITHUB_REF_NAME", os.environ.get("GITHUB_HEAD_REF"))
     else:
-        return (
-            subprocess.run(
-                ["git", "rev-parse", "--symbolic-full-name", "--abbrev-ref", "HEAD"],
-                stdout=subprocess.PIPE,
+        try:
```

Comment:
@fvankrieken Should we let this exception be raised, instead of return `None`? It's kinda disappointing that `mypy` isn't smart enough to flag something like this:
`s3_subpath=Path(git.branch()) / export_type / year,...)`
when git.branch() -> None